Documents/element_design.md

```diff
@@ -0,0 +1,46 @@
+    - [JDI-Light](https://jdi-docs.github.io/jdi-light/?java#theory)
+    - [Bootstrap](https://getbootstrap.com/docs/5.3/components/) 
+    - [Material 2](https://m2.material.io/components?platform=web)
+    - [Material 3](https://m3.material.io/components?platform=web)
+
+- **Requirement Analysis:** Determine the functionalities and states this element can have. For instance, a button may have different states like hover, disabled, or pressed. For this step it is also good to use Material Design guidelines referencesd above.
+
+### 2: Define the Interface
+- **List Properties:** Decide on the properties your element should expose. For a button, properties could include text, color, size, and state.
+- **Define Methods:** Outline the actions you can perform with the element, e.g., click, getText, isEnabled.
+- **Consider Events:** Think about what events the element should respond to, e.g., onClick, onFocus.
+
+### 3: Consider Inheritance and Abstraction
+- **Base Classes:** Look for any existing elements in JDI-Light that are similar and can act as base classes. This helps in reusing code and keeping the design DRY (Don't Repeat Yourself).
+- **Abstract Elements:** If needed, design an abstract class that holds common properties and methods for a category of elements. For example, all form controls might share certain behaviors.
+
+### 4: Incorporate Element Locators
+- **Locators:** Define how the element will be located in the DOM. Choose the most stable locators to ensure reliable tests.
+- **Methods for Locators:** Decide how the element can change its locators if needed, to make the custom element flexible and reusable in different contexts.
+
+### 5: Design for extensibility
+- Consider how the element can be customized or extended. Users of your framework might need to adjust the default behavior or appearance.
+
+### 6: Mockups and Prototypes
+- **Visual design:** Draw a visual representation of the element to understand how it should look and behave. Use prototyping platforms like Figma, Sketch, or InVision.
+- **Prototype:** Define example scenarios where the UI element would be used, what the expected user input and system responses would be. This can include common and edge case scenarios. Write detailed UI specifications that describe the functionality, appearance, and behavior of the element. These should include dimensions, color schemes, typographic details, and states (e.g., hover, disabled).
+
+### 7: Document the Design
+- **Documentation:** Write clear documentation outlining how the element is supposed to work and how it should be used within the JDI-Light framework.
+- **Examples:** Provide examples of usage including variations of the element if applicable.
+- **Javadoc** Write all documentation related to an element in a form of Javadoc inside an element class.
```

Comment:
and do not forget about JDIAction annotation

Reply:
added